src/components/AddressSearch.js

```diff
@@ -0,0 +1,124 @@
+                .find(component => _.contains(component.types, 'locality'))
+                .get('long_name')
+                .value();
+            const state = _.chain(details.address_components)
+                .find(component => _.contains(component.types, 'administrative_area_level_1'))
+                .get('short_name')
+                .value();
+            const zipCode = _.chain(details.address_components)
+                .find(component => _.contains(component.types, 'postal_code'))
+                .get('long_name')
+                .value();
+
+            // Trigger text change events for each of the individual fields being saved on the server
+            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
+            props.onChangeText('addressCity', city);
+            props.onChangeText('addressState', state);
+            props.onChangeText('addressZipCode', zipCode);
+        }
+    };
+
+    return (
+        <GooglePlacesAutocomplete
+            ref={ref}
+            fetchDetails
+            keepResultsAfterBlur
+            suppressDefaultStyles
+            enablePoweredByContainer={false}
+            onPress={(data, details) => saveLocationDetails(details)}
+            query={{
+                key: 'AIzaSyC4axhhXtpiS-WozJEsmlL3Kg3kXucbZus',
+                language: props.preferredLocale,
+            }}
+            requestUrl={{
+                useOnPlatform: 'web',
+                url: `${CONFIG.EXPENSIFY.URL_EXPENSIFY_COM}api?command=Proxy_GooglePlaces&proxyUrl=`,
+            }}
+            textInputProps={{
+                InputComp: ExpensiTextInput,
+                label: props.label,
+                containerStyles: props.containerStyles,
+            }}
+            styles={{
+                textInputContainer: [styles.googleSearchTextInputContainer],
+            }}
+            renderRow={(data, index) => {
+                // This is using a custom render component in order for the styles to be properly applied to the top row.
+                // The styles for the top and bottom row could have instead by passed to `styles.listView` for
+                // <GooglePlacesAutocomplete>, however the list is always visible, even when there are no results.
+                // Because of this, if the padding and borders are applied to the list, even when it's empty, it takes
+                // up space in the UI and looks like a horizontal line with padding around it.
+                // Using this custom render, the rounded borders and padding can be applied to the first row
+                // so that they are only visible when there are results.
+                const rowStyles = [styles.pv4, styles.ph3, styles.borderLeft, styles.borderRight, styles.borderBottom];
+
+                if (index === 0) {
+                    rowStyles.push(styles.borderTop);
+                    rowStyles.push(styles.mt2);
+                }
```

Comment:
Are we able to do some :first-child css pseudo class here?

Reply:
I looked into that, because I had the same thought, and no... React native styles don't give us that functionality :( IF ONLY THERE WAS A WAY TO KNOW HOW MANY RESULTS THERE WERE!!!! We could fork the component and add that into it, but I couldn't find any other way.